compiler/rustc_target/src/abi/call/mod.rs

```diff
@@ -255,11 +255,14 @@ pub struct Uniform {
     ///   can be shorter, i.e., `{ i64, i64, i32 }` for
     ///   64-bit integers with a total size of 20 bytes.
     pub total: Size,
+
+    /// Force the use of an array, even if there is only a single element.
+    pub force_array: bool,
```

Comment:
Given that so many places are just passing `false`, it might be nicer to have a `fn new(unit, total)` for non-array and another constructor for arrays.

Reply:
This is done now. All uses switched to either Uniform::new or Uniform::consecutive.